pyqtgraph/parametertree/parameterTypes/calendar.py

```diff
@@ -51,5 +51,6 @@ def _interpretValue(self, v):
     def saveState(self, filter=None):
         state = super().saveState(filter)
         fmt = self._interpretFormat()
-        state['value'] = state['value'].toString(fmt)
+        if state['value'] is not None:
```

Comment:
under what conditions would this be `None`?  Also is there a case where the key `value` would be missing and we should use `state.get('value')` instead?

Reply:
`value` is guaranteed to exist in the state and is `None` if nothing is initially specified.

I see two easy options for a better initial value if the user doesn't specify anything:
- `QDate()` would give a null/invalid (but date-like) value
- `QDate.currentDate()` is what the calendar widget's selected day defaults to. I'd lean toward this one since it's intuitive based on the calendar being displayed. Of course, a minor downside is potentially non-reproducible results overtime due to a changing default value.

Reply:
I think `QDate.currentDate()` is a pretty good default, but I think changing this might be a bit out of scope for this PR.